### ISSUE TITLE
Interaction calls getBoundingClientRect on mousemove events

### ISSUE THREAD

**Issue by tmcw** [mapbox/mapbox-gl-js#854]
`getBoundingClientRect` [triggers reflows](http://gent.ilcore.com/2011/03/how-not-to-trigger-layout-in-webkit.html) because it requires style recalculation. It's best not to call this function on fast-firing event handlers.

**Comment by bhousel** [comment]
@mourner Can you review 4e19ca2 when you get a chance?

**Comment by bhousel** [comment]
Just putting a quick update here:  I did try caching the offsets on `mouseenter` of the canvas container, but it caused some issues when there were multiple maps on the same page and they were both trying to reset the cache.

More info here:  https://github.com/mapbox/mapbox-gl-js/pull/1598#issuecomment-158270377

@jfirebaugh and I chatted and we both agree that any minor performance bump this trick might get us is not worth the complexity of introducing a cache, and we can't think of a way to avoid calling `getBoundingClientRect`.

e.g. offset adjustment in the middle of a `mousemove`:
![cached offset issue](https://cloud.githubusercontent.com/assets/38784/11293234/263c0e84-8f26-11e5-9799-d9c756fe9027.gif)

Closing this issue as "wontfix" for now..
